homeassistant/components/dremel_3d_printer/sensor.py

```diff
@@ -0,0 +1,134 @@
+"""Support for monitoring Dremel 3D Printer sensors."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import Dremel3DPrinterDataUpdateCoordinator, Dremel3DPrinterDeviceEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the available Dremel 3D Printer sensors."""
+    coordinator: Dremel3DPrinterDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    device_id = config_entry.unique_id
+
+    assert device_id is not None
+
+    entities: list[SensorEntity] = [
+        Dremel3DPrinterStatusSensor(coordinator, config_entry),
+        Dremel3DPrinterProgressSensor(coordinator, config_entry),
+        Dremel3DPrinterTemperatureSensor(coordinator, config_entry, "chamber", False),
+        Dremel3DPrinterTemperatureSensor(coordinator, config_entry, "platform"),
+        Dremel3DPrinterTemperatureSensor(coordinator, config_entry, "extruder"),
+    ]
+
+    async_add_entities(entities)
```

Comment:
Please have a look at the codebase for examples on how to use `SensorEntityDescription`.